openapi_codec/encode.py

```diff
@@ -23,15 +24,61 @@ def generate_swagger_object(document):
         swagger['schemes'] = [parsed_url.scheme]
 
     swagger['paths'] = _get_paths_object(document)
+    swagger['definitions'] = _get_definitions(document)
 
     return swagger
 
 
+def _get_definitions(document):
+
+    definitions = OrderedDict()
+    links = _get_links(document)
+
+    def get_field_def_data(field_item, defs):
+
+        definition_data = {
+            'type': 'object',
+            'properties': {}
+        }
+
+        if isinstance(field_item, coreschema.Object):
+            props = field_item.properties
```

Comment:
since the recursion can also be an array of "simple" types, i would also suggest to add here:

```python
        elif isinstance(field_item, (coreschema.String, coreschema.Boolean, coreschema.Integer, coreschema.Number)):
            return {'type': _get_field_type(field_item), 'description': _get_field_description(field_item)}
```